### DISCUSSION TITLE
Image Optimization for External Domains: Support URL patterns for subdomains

### DISCUSSION THREAD

**Discussion by 0xjessel** [vercel/next.js#25174] [Ideas]
### Describe the feature you'd like to request

# Request: The ability to support multiple subdomains via regex/patterns for Next.js default image optimization.  

## Problem 
I'm building out a blog/personal website.  I would like to use `next/image` to render my Instagram photos but Instagram's API serves thumbnail URLs like [this](https://scontent.cdninstagram.com/v/t51.29350-15/118880713_151718926592744_1865537361354797068_n.jpg?_nc_cat=109&ccb=1-3&_nc_sid=8ae9d6&_nc_ohc=_IZH7C400xQAX9xt9mF&_nc_ht=scontent.cdninstagram.com&oh=59d1c00113a31c88bfe9e6337a96d934&oe=60C73CD7) where the hostname is `scontent.cdninstagram.com`.  Great, so I add this hostname to my `next.config.js` to whitelist this external domain for image optimization.  

However, Instagram has CDNs with different subdomains depending on region and so when I push my deployment on vercel, it's fetching my Instagram photos from [here](https://scontent-iad3-2.cdninstagram.com/v/t51.29350-15/118880713_151718926592744_1865537361354797068_n.jpg?_nc_cat=109&ccb=1-3&_nc_sid=8ae9d6&_nc_ohc=_IZH7C400xQAX80kYr0&_nc_ht=scontent-iad3-2.cdninstagram.com&oh=26d62b260043a3a705f1e6387ec200be&oe=60C73CD7&w=3840&q=75) where the hostname is `scontent-iad3-2.cdninstagram.com`.  

Therein lies the problem, the whitelist external domains in my `next.config.js` cannot possibly enumerate all the possible CDNs that Instagram has.  A quick google search shows at least 20+ which is bound to change over time and is not exhaustive.  

This problem arises in any situation (not just IG) where you're trying to render images on Next.js from a global CDN service that have different subdomains.  

![image](https://user-images.githubusercontent.com/394218/118396497-85b36280-b5eb-11eb-9291-5e22ad4f1717.png)

### Describe the solution you'd like

Some ability to patternize URL matching.  In the example above, I could whitelist all of Instagram's CDN with something like:

```javascript
module.exports = {
  images: {
    domains: ['*.cdninstagram.com'],
  },
}  
```

### Describe alternatives you've considered

I can consider not using `next/image` for my `img` component - but was hoping to take advantage of the image optimization that Next.js provides.   

I can also consider using the `loader` functionality to have a third-party service handle it - most services cost money though and that'd be a heavy lift just to display some Instagram photos.  

Hopefully I'm not missing something obvious!

**Comment by TomAstley** [comment]
I'm having the same issue. Until another solution becomes available, I am aware of the following sub-domains for Instagram. Hope this helps others. I'm sure there many more sub-domains I am missing!

scontent-ams3-1.cdninstagram.com
scontent-amt2-1.cdninstagram.com
scontent-arn2-1.cdninstagram.com
scontent-atl3-1.cdninstagram.com
scontent-dft4-1.cdninstagram.com
scontent-dft4-2.cdninstagram.com
scontent-dfw5-1.cdninstagram.com
scontent-dfw5-2.cdninstagram.com
scontent-frt3-1.cdninstagram.com
scontent-frx5-1.cdninstagram.com
scontent-iad3-1.cdninstagram.com
scontent-iad3-2.cdninstagram.com
scontent-hou1-1.cdninstagram.com
scontent-kul1-1.cdninstagram.com
scontent-lga3-2.cdninstagram.com
scontent-mia1-1.cdninstagram.com
scontent-mia3-1.cdninstagram.com
scontent-mia3-2.cdninstagram.com
scontent-msp1-1.cdninstagram.com
scontent-ort2-1.cdninstagram.com
scontent-ort2-2.cdninstagram.com
scontent-sea1-1.cdninstagram.com
scontent-sit4-1.cdninstagram.com
scontent-sjc2-1.cdninstagram.com
scontent-syd2-1.cdninstagram.com
scontent.cdninstagram.com

**Reply by shonsirsha** [reply]
Thanks a lot!!

**Comment by florianbepunkt** [comment]
Has this been solved?

**Comment by Pablo-Conte** [comment]
Hey guys! the problem was been solved?

**Reply by CarrettaRiccardo** [reply]
Seems like the * wildcard is working now, not sure from which version of nexjs. I am testing with `next@14.2.2`:

```
  images: {
    remotePatterns: [
      {
        protocol: "https",
        hostname: "*.cdninstagram.com",
        port: "",
      },
    ],
  },
```
